### PR TITLE
Upgrades master branch to use clang-format v15.0.7

### PR DESCRIPTION
**Pull request by zwoop**
In addition to the upgrade to v15.0.7, with its new features and fixes, this also adds binaries for aarch64 platforms for both Linux and macOS.

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci]

**Comment by zwoop** [comment]
[approve ci]

**Review by maskit — COMMENTED** [review]
👍 Tested the binaries on aarch64.

File: iocore/dns/SplitDNS.cc [hunk 1]
```diff
@@ -407,7 +407,7 @@ SplitDNSRecord::ProcessDomainSrchList(char *val)
 
   for (int i = 0; i < numTok; i++) {
     const char *current = pTok[i];
-    int cnt             = sz += strlen(current);
+    int cnt = sz += strlen(current);
```
Comment (bneradt) [inline iocore/dns/SplitDNS.cc:410]: This is an interesting change. Most of the other changes add alignment with the previous `=` or whatever, this removes it. (This isn't a problem, just something that caught my eye.)

**Review by bneradt — APPROVED** [review]
I skimmed through about half of these files, up until github was autohiding the files because of the volume. I then ignored white space changes, which highlighted the changes that added or removed newlines.

I didn't see any changes I didn't like. I think these changes are an improvement and will make things more readable.

Thanks for keeping clang-format updated.

**Comment by bneradt** [comment]
And I do definitely like having the arm64 clang-format binaries. That makes running clang-format much quicker on my system.

File: iocore/dns/SplitDNS.cc (hunk 1 above)
Comment (zwoop) [reply iocore/dns/SplitDNS.cc:410]: Yeh, there are still imperfections :).

**Review by bryancall — APPROVED** [review]

**Pull request merged**